extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/interceptor/TransactionalInterceptorBase.java

```diff
@@ -86,8 +104,21 @@ protected Object invokeInOurTx(InvocationContext ic, TransactionManager tm) thro
     protected Object invokeInOurTx(InvocationContext ic, TransactionManager tm, RunnableWithException afterEndTransaction)
             throws Exception {
 
-        tm.begin();
-        Transaction tx = tm.getTransaction();
+        TransactionConfiguration configAnnotation = getTransactionTimeout(ic);
+        if (configAnnotation != null) {
+            tm.setTransactionTimeout(configAnnotation.timeout());
+        }
+        Transaction tx;
+        try {
+            tm.begin();
+            tx = tm.getTransaction();
+        } finally {
+            if (configAnnotation != null) {
+                //restore the default behaviour
+                tm.setTransactionTimeout(0);
```

Comment:
I expect it would want to setTransactionTimeout to the value of a new call of tm.getTransactionTimeout() that is issued before the set of the config value done on line 108 added above

Reply:
That's likely a bit better and would cover the case where a user manually uses the `tm.setTransactionTimeout()` herself. But in that situation we might be screwed if the user uses that API within a `@TransactionConfiguration` marked method. Unless we keep the old value of 108 around and before reapplying it, you check that the new value is not different from the timeout one. There is still a window of bug but smaller.

Reply:
Yeah, this is true. So we get the timeout before the TX starts, then get it again after and reset to either: 1. the first one if the second one is equal, the second one if the first one is not equal to it and it is different to the one from the annotation, 3. nothing if first one and the annotation match (for completeness)

Reply:
I'm agree that we need to restore to the previous transaction manager timeout, but I'm not sure it's possible that between the time I get the timeout and the time I set it back it can be changed because the timeout is set at Thread level and the Thread is in the same method.
So for me, no need to a more complex implementation that 
1. Get the transaction manager current timeout
2. Set the custom timeout for this transaction
3. Restore the transaction manager timeout to it's previous value

Reply:
The algorithm proposed by Tom covers almost all cases. More than your proposed algorithm.

- read_annotation_timeout
- read_timout_before_tx
- read_timout_after_tx
- if (read_timeout_after_tx != read_annotation_timeout (if set) ) leave it (it has been overridden by the user)
- else setTimeout to read_timout_before_tx

Reply:
As the setTImeout occurs before the interceptor invoke the real method, I cannot think it's possible that someone update the timeout of the transaction manager between the get and the set. Anyway if you really want I can implement this more complex algorithm.

Reply:
+1, unless .begin does something strange

Reply:
Ah yes, sorry. I got confused that thought that the the reset happened after the real method was called. My bad. Forget what I said.

Reply:
I check the begin method and it does nothing but create a transaction and put it in a map so my algotithm is safe. So I propose to keep it as is, it's simplier (and simplier is better :) )

Reply:
I suppose it could depend on the vendors implementation of .begin but I do think it is very unlikely for a vendor to have made an implementation choice that affects the timeout